atom/browser/extensions/atom_extension_system.cc

```diff
@@ -0,0 +1,185 @@
+// Copyright 2014 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "atom/browser/extensions/atom_extension_system.h"
+
+#include <memory>
+#include <string>
+
+#include "atom/browser/extensions/atom_extension_loader.h"
+#include "base/bind.h"
+#include "base/files/file_path.h"
+#include "base/files/file_util.h"
+#include "base/task/post_task.h"
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/browser_task_traits.h"
+#include "content/public/browser/browser_thread.h"
+#include "content/public/browser/notification_details.h"
+#include "content/public/browser/notification_service.h"
+#include "content/public/browser/notification_source.h"
+#include "extensions/browser/api/app_runtime/app_runtime_api.h"
+#include "extensions/browser/extension_registry.h"
+#include "extensions/browser/info_map.h"
+#include "extensions/browser/notification_types.h"
+#include "extensions/browser/null_app_sorting.h"
+#include "extensions/browser/quota_service.h"
+#include "extensions/browser/runtime_data.h"
+#include "extensions/browser/service_worker_manager.h"
+#include "extensions/browser/shared_user_script_master.h"
+#include "extensions/browser/value_store/value_store_factory_impl.h"
+#include "extensions/common/constants.h"
+#include "extensions/common/file_util.h"
+
+using content::BrowserContext;
+using content::BrowserThread;
+
+namespace extensions {
+
+AtomExtensionSystem::AtomExtensionSystem(BrowserContext* browser_context)
+    : browser_context_(browser_context),
+      store_factory_(new ValueStoreFactoryImpl(browser_context->GetPath())),
+      weak_factory_(this) {}
+
+AtomExtensionSystem::~AtomExtensionSystem() = default;
+
+const Extension* AtomExtensionSystem::LoadExtension(
+    const base::FilePath& extension_dir) {
+  return extension_loader_->LoadExtension(extension_dir);
+}
+
+const Extension* AtomExtensionSystem::LoadApp(const base::FilePath& app_dir) {
+  return LoadExtension(app_dir);
```

Comment:
Maybe we should `CHECK` here -- maybe we don't want to support this?

Reply:
For this PR, I did intentionally remove some code related to loading app extensions. If this function doesn't require an implementation, it should be removed.